### PR TITLE
Add generated and hosted docs

### PR DESCRIPTION
**Pull request by sdl1**
Addresses https://github.com/ScottishCovidResponse/SCRCIssueTracking/issues/204

- I put the existing docs into Documenter.jl format. To build and view these locally:
```
julia --project=docs/ -e 'using Pkg; Pkg.develop(PackageSpec(path=pwd())); Pkg.instantiate()'
julia --project=docs/ docs/make.jl
chrome docs/build/index.html
```
- Added a travis job to build the docs (just following the [Documenter docs](https://juliadocs.github.io/Documenter.jl/stable/man/hosting/#Travis-CI-1) here)
- I set it up so the "dev" docs should be deployed on commits to the dev branch (`devbranch = "dev"` in `deploydocs`), since this is the default branch at the moment. We might want to change this to `master` later on

To make the deployment work there are two things we need to do which I don't have permissions for, could @ScottishCovidResponse/simulation-jl-admins  do these please?
1. Set up a deploy key as explained [here](https://juliadocs.github.io/Documenter.jl/stable/man/hosting/#travis-ssh-1)
2. In the GitHub repo settings, enable GitHub pages on the `gh-pages` branch (which I think should be created with the first successful deployment, so we might need to do this after 1. and after this PR is merged)

### CODE REVIEW

**Review by richardreeve — COMMENTED** [review]
Looks great. Unless I'm missing something, don't we need a link from the `README.md` though before we merge it?

File: docs/Project.toml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[deps]
+Documenter = "e30172f5-a6a5-5a46-863b-614d45cd2de4"
+
```
Comment (richardreeve) [inline docs/Project.toml:4]: And don't the docs have a dependency on `Simulation` too given that you're instantiating this folder? I'm not really sure tbh...

Comment (sdl1) [reply docs/Project.toml:4]: Yes good spot, added.

**Comment by sdl1** [comment]
Added badge-links for docs and CI in the README. (Both for dev branch at the moment)

**Review by richardreeve — APPROVED** [review]

**Comment by richardreeve** [comment]
It seems to be erroring on continuous-integration/travis-ci/push, but assuming that passes now I've restarted it, that looks good to go.

**Comment by richardreeve** [comment]
Hmm. The documentation CI seem to be failing which is obviously a problem.

**Comment by sdl1** [comment]
> Hmm. The documentation CI seem to be failing which is obviously a problem.

Has passed now (there were some issues yesterday with julialang.org being down which I think caused a bunch of pipeline failures)

**Comment by richardreeve** [comment]
@claireh93 Do you want to merge this now? I'm happy with it...

**Comment by sdl1** [comment]
Just a reminder about the bits to make the deployment work (I think this should be all...):
>1. Set up a deploy key as explained [here](https://juliadocs.github.io/Documenter.jl/stable/man/hosting/#travis-ssh-1)
>2. In the GitHub repo settings, enable GitHub pages on the gh-pages branch (which I think should be created with the first successful deployment, so we might need to do this after 1. and after this PR is merged)

**Comment by claireh93** [comment]
Sure, am I able to add the deploy key or is that something only you have permission for @richardreeve? Looks great though, thanks Sean!

**Comment by richardreeve** [comment]
I have no idea. I'll try doing it now though...

**Comment by richardreeve** [comment]
Okay, fingers crossed I got that right.

**Comment by sdl1** [comment]
Cool, I think it should be added to the github settings page rather than the repo though. The settings page should be accessible at https://github.com/ScottishCovidResponse/Simulation.jl/settings/keys, and there should be a button there which says 'Add deploy key'?

**Comment by richardreeve** [comment]
Oops. Hopefully this time it has worked!

**Comment by richardreeve** [comment]
Okay, @sdl1 - do you want to merge this if you think it's fine this time?

**Pull request merged**